### PR TITLE
PIM-4308: Command to ensure MongoDB indexes

### PR DESCRIPTION
**Pull request by fitn**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Y
| New feature?         | Y
| BC breaks?           | Y
| CI currently passes? | ?
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N (but already broken)
| PMD issues?**        | N (but already broken)
| Changelog updated?   | Y
| Fixed tickets        | PIM-4308
| DB schema updated?   | N
| Migration script?    | N
| Doc PR               | N

CI : http://ci.akeneo.com/job/pim-pr-behat/2467/
http://ci.akeneo.com/job/pim-pr-behat-mongodb/788/

This ticket needs to be merge on master too. Just remove default value for $attributeClass parameter in IndexCreator.

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Command/MongoDBIndexCreatorCommand.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('pim:product:ensure-mongodb-indexes')
+            ->setDescription(
+                'Ensure MongoDB indexes of the products '
+                .'(completeness, unique attributes and attributes used in the grid) are created'
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Command/MongoDBIndexCreatorCommand.php]: please put the point on the previous line and a space before it

File: src/Pim/Bundle/CatalogBundle/Command/MongoDBIndexCreatorCommand.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+            return -1;
+        }
+
+        $indexCreator = $this->getIndexCreator();
+        $indexCreator->ensureUniqueAttributesIndexes();
+        $indexCreator->ensureCompletenessesIndexes();
+        $indexCreator->ensureAttributesIndexes();
+
+        return;
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Command/MongoDBIndexCreatorCommand.php]: you should return a numerical value as you do 7 lines above :)

**Comment by sumbobyboys** [comment]
Nice PR and thanks for the constant :)

**Comment by sumbobyboys** [comment]
GTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,10 @@
 ## Bug fixes
 - PIM-4227: fix BC break introduced in 1.3.13
 - PIM-4309: Fix bug processing media with a non existing media
+- PIM-4308: MongoDB indexes are removed on schema update
+
+## BC breaks
+- Change constructor of `Pim\Bundle\CatalogBundle\Doctrine\MongoDBODM\IndexCreator`, added Attribute class name
```
Comment (nidup) [inline CHANGELOG.md]: no BC break in 1.3 ;)

here you define a default value in the constructor so it's a BC only if someone has overriden this class and doesn't call the parent constructor in that case we could define the default value in protected $attributeClass = 'mydefaultvalue' and it should solve the case

**Pull request merged**